lib/src/generators/router/generator/router_generator.dart

```diff
@@ -38,13 +41,25 @@ class RouterGenerator implements BaseGenerator {
       routes: _rootRouterConfig.routesIncludingTheirChildren,
     ).build(emitter);
 
+    /// Remove duplicate classes from nested routers
+    List<String> classesNames = [];
+    List<Spec> parsedClasses = [];
```

Comment:
It's possible for us to remove the `classNames` from this implementation.

If we change 

```dart
if (classesNames.contains(c.name)) continue;
```

to 

```dart
if (parsedClasses.any((parsedClass) => parsedClass.name == c.name)) continue;
```

Then we wouldn't need the `classesNames` list anymore.

Reply:
@FilledStacks  You are right, but if you try the condition above you will get the following error.

`The getter 'name' isn't defined for the type 'Spec'.`

Reply:
But... I changed the List to be of type `Class` instead of `Spec` and we have no problem. Ready to merge my friend 😉